### PR TITLE
Alasdair/ecom 2958 publish validation

### PR DESCRIPTION
**Pull request by AlasdairSwan**
@jimabramson @rlucioni please review.

### CODE REVIEW

File: programs/static/js/views/program_details_view.js [hunk 1]
```diff
@@ -145,6 +153,27 @@ define([
...
+                    $message = {};
+
+                if ( this.model.get( 'marketing_slug' ).length > 0 ) {
+                    isValid = true;
+                } else {
+                    $input = this.$el.find( '#program-marketing-slug' );
+                    $message = $input.siblings( '.field-message' );
+
+                    // Update DOM
+                    $input.addClass( 'has-error' );
```
Comment (rlucioni) [inline programs/static/js/views/program_details_view.js]: Should we be using `errorClass` from `validation_config.js`?

File: programs/static/js/views/program_details_view.js [hunk 2]
```diff
@@ -145,6 +153,27 @@ define([
...
+                    $message.addClass( 'has-error' );
+                    $message.find( '.field-message-content' )
```
Comment (rlucioni) [inline programs/static/js/views/program_details_view.js]: Similarly, should this be `messageContent` from `validation_config.js`?

Comment (rlucioni) [inline programs/static/js/views/program_details_view.js]: And on a related note, am I correct in saying that Backbone.Validation is clearing these error classes when validation succeeds?

Comment (AlasdairSwan) [inline programs/static/js/views/program_details_view.js]: Yes it is clearing the message. I will check whether messageContent is available here, if so I will use it.

File: programs/static/js/views/program_details_view.js (hunk 1 above)
Comment (AlasdairSwan) [inline programs/static/js/views/program_details_view.js]: These variables are scoped to the validation_config.js so are not available here.

File: programs/static/js/views/program_details_view.js (hunk 2 above)
Comment (AlasdairSwan) [inline programs/static/js/views/program_details_view.js]: The variables are scoped to the validation_config.js so are not available here.

Comment (AlasdairSwan) [inline programs/static/js/views/program_details_view.js]: Also, as this field and validation are a short term fixed I'm not too worried about having the error classes defined in multiple places.

**Comment by rlucioni** [comment]
:+1:

**Comment by jimabramson** [comment]
LGTM :+1:

**Pull request merged**